dashboard/generate.py

```
@@ -137,6 +139,36 @@ def efu_get_version(self, ipaddr, port):
             self.dprint("connection reset (by peer?)")
             return "connection reset (by peer?)"
 
+
+    def check_fw_pipeline(self, ipaddr, port):
+        if self.test:
+            return 5
+        try:
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            s.connect((ipaddr, port))
+            s.shutdown(socket.SHUT_WR) # no data sent
+            data = b""
+            while True:
+                chunk = s.recv(4096)
+                if not chunk:
+                    break
+                data += chunk
+            lines = data.decode("utf-8", errors="ignore").strip().splitlines()
+            for line in lines:
+                try:
+                    obj = json.loads(line)
+                    # Look for any key ending with '.worker_state'
+                    for key, value in obj.items():
+                        if key.endswith(".worker_state"):
+                            return int(value)
+                except Exception:
+                    continue
+                return 0  # Not found
+        except Exception:
+            self.dprint("connection reset (by peer?)")
+            return 0
```

I think in this case the exception handling might be hiding the real error and just assuming the connection was reset. It could make it harder to understand what actually went wrong.

We should at least handle [OSError](https://docs.python.org/3/library/exceptions.html#OSError) and [ConnectionError](https://docs.python.org/3/library/exceptions.html#ConnectionError)

```
except ConnectionError:
    print("Connection reset by peer")
except OSError as e:
    print(f"Socket error: {e}")
```

about workerState=0. does that mean the worker is idling? If the data is missing or couldn’t be parsed, maybe using something like -1 would make it clearer that the value is unknown, rather than a valid state.

---

`ConnectionError` is a subclas of `OSError` so catching only `OSError` is just fine.